javatests/arcs/core/storage/StoreWriteBackTest.kt

```diff
@@ -140,7 +180,62 @@ class StoreWriteBackTest {
         assertThat(output).isEqualTo(TEST_RANGE.toList())
     }
 
+    @Test
+    fun dataVersionInOrder() = runBlocking {
+        val versions = arrayListOf<Int>()
+        databaseFactory.addClients(
+            object : DatabaseClient {
+                override val storageKey = testKey.storageKey
+                override suspend fun onDatabaseUpdate(
+                    data: DatabaseData, version: Int, originatingClientId: Int?
+                ) = synchronized(versions) {
+                    versions.add(version)
+                    Unit
+                }
+                override suspend fun onDatabaseDelete(originatingClientId: Int?) = Unit
+            }
+        )
+
+        val refModeStore = createReferenceModeStore()
+        for (i in 1..NUM_OF_WRITES) {
+            refModeStore.onProxyMessage(
+                ProxyMessage.Operations(
+                    listOf(
+                        RefModeStoreOp.SetAdd(
+                            "me",
+                            VersionMap("me" to i),
+                            createEmptyPersonEntity("e$i")
+                        )
+                    ),
+                    id = 1
+                )
+            )
+        }
+
+        refModeStore.containerStore.awaitIdle()
+        assertThat(versions.toList()).isEqualTo((1..NUM_OF_WRITES).toList())
+    }
```

Comment:
This is a great integration test of RefModeStore, DirectStore, and the WriteBack. I feel like it might better reside in a ReferenceModeStoreTest class, or one of its variants.  Also, it could be nice to have a more narrowly-scoped unit test of DirectStore's usage of the WriteBack... but that could wait until a later date.

Reply:
Sounds good!

My intention was to use the integration test to detect versioning and ordering problems involving deeper and fuller stack. A more narrowly-scoped test would absolutely help narrow down where a problem actually comes from. We can do it at later PRs.

I was also thinking to create more multi-threading integration tests down the road after all Arcs recipes will become richer and complex over time, and more clients will engage i.e. now fedex uses Arcs storage for logging, more clients may do the same logging via Arcs one day and some of them could even race writing to the same storage, etc.